src/plugin/transform.ts

```diff
@@ -92,19 +100,21 @@ export function transform({
         mdDir.renderer.rules[TOKEN_TYPE] = (tokens, idx, _opts, _env, self) => {
             const token = tokens[idx];
 
-            const base = `<base target="${baseTarget}">`;
-            let content = base + token.attrGet(TokenAttr.srcdoc) ?? '';
-
+            let additional = baseTarget ? `<base target="${baseTarget}">` : '';
             if (styles) {
                 const stylesContent =
                     typeof styles === 'string'
                         ? `<link rel="stylesheet" href="${styles}" />`
                         : `<style>${getStyles(styles)}</style>`;
-                content = stylesContent + content;
+                additional += stylesContent;
             }
 
-            const resultContent = sanitize ? sanitize(content) : content;
-            token.attrSet(TokenAttr.srcdoc, resultContent);
+            const head = `<head>${headContent || additional}</head>`;
+            const body = `<body>${token.attrGet(TokenAttr.srcdoc) ?? ''}</body>`;
+            const html = `<!DOCTYPE html><html>${head}${body}</html>`;
```

Comment:
Right, I think that eliminating quirks mode (as we discussed off the hub) is the right move here. This seems good.